src/main/java/spike/storage/TaskList.java

```diff
@@ -0,0 +1,157 @@
+package spike.storage;
+
+import spike.exceptions.SpikeException;
+import spike.tasks.Task;
+import spike.tasks.Deadline;
+import spike.tasks.Event;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the user's list of tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Constructs an empty TaskList.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Constructs a TaskList with the given tasks.
+     *
+     * @param tasks The tasks to be added to the TaskList.
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Returns a copy of the list of tasks.
+     *
+     * @return A copy of the list of tasks.
+     */
+    public ArrayList<Task> getAllTasks() {
+        return new ArrayList<>(this.tasks);
+    }
+
+    /**
+     * Returns the number of tasks in the list.
+     *
+     * @return The number of tasks in the list.
+     */
+    public int getSize() {
+        return tasks.size();
+    }
+
+    /**
+     * Returns the description of the task at the specified index.
+     *
+     * @param index The index of the task whose description is to be returned.
+     * @return The description of the task at the specified index.
+     * @throws IndexOutOfBoundsException If the index is out of bounds.
+     */
+    public String getTaskString(int index) throws IndexOutOfBoundsException {
```

Comment:
```suggestion
    public String getTaskDescription(int index) throws IndexOutOfBoundsException {
```
While `getTaskString` suffice, it would be better to use `getTaskDescription` to make it intuitive that this method is specifically getting the description of the task.